.github/workflows/deploy-docusaurus.yml

```diff
@@ -0,0 +1,33 @@
+name: Deploy to GitHub Pages
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - 'docusaurus/**'
```

Comment:
@relsteiner Thought - Shall we maybe put this under ~root/docs/documentation-site so all our docs are centralized in a single folder?

Reply:
I thought that with time `/docuzaurus` will become our single documentation folder (already migrated the decisions folder from `/docs` to `/docuzaurus`).

Reply:
@relsteiner Consider the following scenario - A developer is visiting the repo, not the website (many prefer GitHub vs custom site), she is looking to read some contribution/tech docs, she inspects the repo folders for a doc/docs/documentation folder but there is not any... She only sees this weird word 'docusaurus' and believe it contains something irrelevant

Not critical for now, down the road I would consider changing